TTHAnalysis/python/tools/higgsDiffGenTTH.py

```diff
@@ -30,43 +35,87 @@ def beginFile(self, inputFile, outputFile, inputTree, wrappedOutputTree):
         self.out.branch('%snQFromWFromT'%self.label     , 'I')
         self.out.branch('%snLFromWFromH'%self.label     , 'I')
         self.out.branch('%snLFromWFromT'%self.label     , 'I')
+        self.out.branch('%snTauFromH'%self.label        , 'I')
+        self.out.branch('%snhFromTauFromH'%self.label   , 'I')
+        self.out.branch('%snLFromTauFromH'%self.label   , 'I')
+        self.out.branch('%snNuFromTauFromH'%self.label  , 'I')
+        self.out.branch('%snTNuFromTauFromH'%self.label , 'I')
+        self.out.branch('%snZFromH'%self.label          , 'I')
+        self.out.branch('%snQFromZFromH'%self.label     , 'I')
+        self.out.branch('%snLFromZFromH'%self.label     , 'I')
+        self.out.branch('%snNuFromZFromH'%self.label    , 'I')
+        self.out.branch('%snTauFromWFromH'%self.label   , 'I')
+        self.out.branch('%snLFromTauFromWFromH'%self.label  , 'I')
+        self.out.branch('%snNuFromTauFromWFromH'%self.label , 'I')
+        self.out.branch('%snhFromTauFromWFromH'%self.label  , 'I')
 
         # Although we expect a maximum of 2 objects per array, we allow for 4 of them to be stored, for safety and later checks
         for suffix in ["_pt", "_eta", "_phi", "_mass"]:
             
-            self.out.branch('%sHiggses%s'%(self.label,suffix)          , 'F', 4, '%snHiggses'%self.label         ) 
```

Comment:
Why did you take out the reference to the number of filled events from the branch?
Is there anything specifically bad in the way the variables were accessed and filled?